### PR TITLE
fix pipeline due to bug for variable set size and removal of 'n'

### PR DESCRIPTION
**Pull request by BardiyaAk**
- **fix bug for running the script to save data due to 'n' removal**
- **fix bug in prompt generation due to variable set size**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: setlexsem/generate/generate_prompts.py [hunk 1]
```diff
@@ -294,8 +294,12 @@ def main(config_file, save_data, overwrite):
     # Sampler/Sets Config
     SET_TYPES = config["SET_TYPES"]
     N = replace_none(config["N"])
-    M_A = replace_none(config["M_A"])
-    M_B = replace_none(config["M_B"])
+    if "M" in config.keys():
```
Comment (ndronen) [inline setlexsem/generate/generate_prompts.py]: Please add a comment about why this is here.

File: setlexsem/generate/generate_sets.py [hunk 2]
```diff
@@ -208,16 +209,9 @@ def make_hps_set(
...
     if set_type == "numbers":
-        # override n if item-length is defined
-        if hp.get("item_len"):
-            n = None
-        else:
-            n = hp["n"]
-
         sampler = BasicNumberSampler(
-            n=n,
+            n=hp["n"] if hp.get("n") else None,
```
Comment (ndronen) [inline setlexsem/generate/generate_sets.py]: I believe line 214 can be just:
```
n=hp.get("n")
```
as `dict.get(key)` returns `None` if the key is not present in the dictionary.

File: setlexsem/utils.py [hunk 3]
```diff
@@ -79,7 +79,7 @@ def get_prompt_file_path(sampler_hp, prompt_hp, random_seed):
         random_seed,
     )
     experiment_folder_structure = os.path.join(
-        sampler_hp["set_type"],
+        sampler_hp["set_types"],
```
Comment (ndronen) [inline setlexsem/utils.py:82]: Did the move from `set_type` to `set_types` result in config files that have `n` (which is just for sampling numbers) even when `set_types = ["words", "numbers"]`, even though `n` is not applicable for words?

File: setlexsem/utils.py [hunk 4]
```diff
@@ -264,8 +264,12 @@ def read_config(
 
     SET_TYPES: List[str] = config["SET_TYPES"]
     N: List[int] = config["N"]
-    MA: List[int] = config["MA"]
-    MB: List[int] = config["MB"]
+    if "M" in config.keys():
```
Comment (ndronen) [inline setlexsem/utils.py]: Please note in a comment that this is for backward compatibility.

**Review by ndronen — APPROVED** [review]
Left a few small suggestions and a question. LGTM.

File: setlexsem/generate/generate_sets.py (hunk 2 above)
Comment (BardiyaAk) [reply setlexsem/generate/generate_sets.py]: Nice! fixed it!

File: setlexsem/utils.py (hunk 3 above)
Comment (BardiyaAk) [reply setlexsem/utils.py:82]: that was an issue, but we also were calling the key "set_type" at times, and sometimes "set_types". I refactors all of it to "set_types" now.

File: setlexsem/generate/generate_prompts.py (hunk 1 above)
Comment (BardiyaAk) [reply setlexsem/generate/generate_prompts.py]: Added it, also i had it to 2 places, i consolidated it to read_config file so removed one of these.

**Pull request merged**